### ISSUE TITLE
Re-write a few un-Pythonic lines

### ISSUE THREAD

**Issue by clintval** [clintval/sample-sheet#21]
- [x] `self.__dict__.get(attr)`
https://github.com/clintval/sample-sheet/blob/87deb18ce63d2af4828222dd6e6362ed29a501a7/sample_sheet/_sample_sheet.py#L259
- [x] There must be a better way to change a class name without needing useless inheritance
https://github.com/clintval/sample-sheet/blob/87deb18ce63d2af4828222dd6e6362ed29a501a7/sample_sheet/_sample_sheet.py#L320-L325
- [x] `return None if len(self.Reads) == 0 else len(self.Reads) == 1`
https://github.com/clintval/sample-sheet/blob/87deb18ce63d2af4828222dd6e6362ed29a501a7/sample_sheet/_sample_sheet.py#L437-L439
- [x] `VALID_ASCII = {string.ascii_letters + string.digits + '-_'}`
https://github.com/clintval/sample-sheet/blob/87deb18ce63d2af4828222dd6e6362ed29a501a7/sample_sheet/_sample_sheet.py#L31-L36

**Comment by PertuyF** [comment]
A few suggestions/remarks:

### **`self.__dict__.get(attr)`**
you can use `collections.defaultdict(lambda: None)` to achieve the same behavior as `self.get(attr, None)`. But after looking to the code, I'm not sure why you need to overload `__dict__`.


### **change a class name**
seems that you don't need that at all, just use
 ```python
self.header = SampleSheetSection()
```
 in `SampleSheet`'s `__init__`


### **`None if len(self.Reads) == 0`**
```python
return None if not self.Reads else len(self.Reads) == 1
```
will do, as `bool(iterable)` will return `False` only if the iterable is empty.

**Comment by clintval** [comment]
Fixed in PR #24
